packages/gatsby/src/schema/node-model.js

```diff
@@ -71,6 +71,7 @@ class LocalNodeModel {
     this._prepareNodesQueues = {}
     this._prepareNodesPromises = {}
     this._preparedNodesCache = new Map()
+    this._typedKeyValueIndexes = new Map() // See redux/nodes.js for usage
```

Comment:
Maybe if you want to be able to inject this cache for tests, provide injection point here? Alternatively provide "noCache" option.

Reply:
Yeah. I somehow thought most of those tests were able to do this but that's clearly not the case. I'll add a setter so we can override it.